src/test/java/teammates/test/cases/webapi/ResetAccountActionTest.java

```diff
@@ -23,7 +28,66 @@ protected String getRequestMethod() {
     @Override
     @Test
     protected void testExecute() {
-        // TODO
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        StudentAttributes student1OfCourse1 = typicalBundle.students.get("student1InCourse1");
+        loginAsAdmin();
+
+        ______TS("Failure case: no parameters supplied");
+
+        ResetAccountAction a = getAction();
+        JsonResult r = getJsonResult(a);
+        MessageOutput response = (MessageOutput) r.getOutput();
+
+        assertEquals("Either student email or instructor email has to be specified.", response.getMessage());
+        assertEquals(HttpStatus.SC_BAD_REQUEST, r.getStatusCode());
+
+        ______TS("Failure case: no course id supplied");
+
+        String[] paramsInsufficient = {
+                Const.ParamsNames.INSTRUCTOR_EMAIL, instructor1OfCourse1.getEmail(),
+        };
+
+        verifyHttpParameterFailure(paramsInsufficient);
+
+        ______TS("typical success case: reset instructor account");
+
+        String[] paramsInstructor = {
+                Const.ParamsNames.INSTRUCTOR_EMAIL, instructor1OfCourse1.getEmail(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
+        };
+
+        a = getAction(paramsInstructor);
+        r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+        response = (MessageOutput) r.getOutput();
+
+        InstructorAttributes instructor = logic.getInstructorForEmail(instructor1OfCourse1.getCourseId(),
+                instructor1OfCourse1.getEmail());
+
+        assertEquals(response.getMessage(), "Account is successfully reset.");
+        assertNotNull(instructor);
+        assertNull(instructor.getGoogleId());
+        assertNull(logic.getInstructorForGoogleId(instructor1OfCourse1.getCourseId(), instructor1OfCourse1.getGoogleId()));
```

Comment:
What is the second parameter for?

Reply:
The original google id before reset? This is to verify the account now cannot be found with original google id after reset.

Reply:
But you are using `assertNull`...

Reply:
using the original google id attempts to find the account will not success, thus returned null, which should be expected?